prov/efa/src/efa_domain.c

```diff
@@ -494,53 +562,15 @@ int efa_domain_open(struct fid_fabric *fabric_fid, struct fi_info *info,
 
 	domain->cache = NULL;
 
-	/*
-	 * Call ibv_fork_init if the user asked for fork support.
-	 */
-	if (efa_fork_status == EFA_FORK_SUPPORT_ON) {
-		ret = -ibv_fork_init();
-		if (ret) {
-			EFA_WARN(FI_LOG_DOMAIN,
-			         "Fork support requested but ibv_fork_init failed: %s\n",
-			         strerror(-ret));
-			goto err_free_info;
-		}
-	}
-
-	/*
-	 * Run check to see if fork support was enabled by another library. If
-	 * one of the environment variables was set to enable fork support,
-	 * this variable was set to ON during provider init.  Huge pages for
-	 * bounce buffers will not be used if fork support is on.
-	 */
-	if (efa_fork_status == EFA_FORK_SUPPORT_OFF &&
-	    efa_check_fork_enabled(*domain_fid))
-		efa_fork_status = EFA_FORK_SUPPORT_ON;
+#ifndef _WIN32
+	ret = efa_initialize_fork_support(*domain_fid);
 
-	if (efa_fork_status == EFA_FORK_SUPPORT_ON &&
-	    getenv("RDMAV_HUGEPAGES_SAFE")) {
-		EFA_WARN(FI_LOG_DOMAIN,
-			 "Using libibverbs fork support and huge pages is not supported by the EFA provider.\n");
-		ret = -FI_EINVAL;
-		goto err_free_info;
+	if (ret) {
+	    EFA_WARN(FI_LOG_DOMAIN, "Failed to initialize fork support %d", ret);
+	    goto err_free_info;
 	}
+#endif
```

Comment:
You're welcome to do whatever in the efa provider, but the common practice in other parts of the code is to avoid ifdefs within a function.  It makes the code difficult to read/maintain.  It's better to wrap the efa_initialize_fork_suppprt() function with an indef, and add an always success/fail version for windows.  Then just call that function always.